pkg/operation/botanist/controlplane/etcd/bootstrap.go

```diff
@@ -1,12 +1,311 @@
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("50m"),
+										corev1.ResourceMemory: resource.MustParse("128Mi"),
+									},
+									Limits: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("300m"),
+										corev1.ResourceMemory: resource.MustParse("512Mi"),
+									},
+								},
+								Ports: []corev1.ContainerPort{{
+									ContainerPort: 9569,
+								}},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		resources = map[string][]byte{
+			"crd.yaml":                crdYAML.Bytes(),
+			"serviceaccount.yaml":     serviceAccountYAML,
+			"clusterrole.yaml":        clusterRoleYAML,
+			"clusterrolebinding.yaml": clusterRoleBindingYAML,
+			"vpa.yaml":                vpaYAML,
+		}
+	)
+
+	if imageVectorOverwrite != nil {
+		configMapImageVectorOverwrite.Data = map[string]string{druidConfigMapImageVectorOverwriteDataKey: *imageVectorOverwrite}
+		configMapImageVectorOverwriteYAML, err = runtime.Encode(codec, configMapImageVectorOverwrite)
+		if err != nil {
+			return err
+		}
+		resources["configmap-imagevector-overwrite.yaml"] = configMapImageVectorOverwriteYAML
+
+		deployment.Spec.Template.Labels["checksum/configmap-imagevector-overwrite"] = utils.ComputeChecksum(configMapImageVectorOverwrite.Data)
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: druidDeploymentVolumeNameImageVectorOverwrite,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: druidConfigMapImageVectorOverwriteName,
+					},
+				},
+			},
+		})
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      druidDeploymentVolumeNameImageVectorOverwrite,
+			MountPath: druidDeploymentVolumeMountPathImageVectorOverwrite,
+			ReadOnly:  true,
+		})
+		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  imagevector.OverrideEnv,
+			Value: druidDeploymentVolumeMountPathImageVectorOverwrite + "/" + druidConfigMapImageVectorOverwriteDataKey,
+		})
+	}
+
+	deploymentYAML, err := runtime.Encode(codec, deployment)
+	if err != nil {
+		return err
+	}
+	resources["deployment.yaml"] = deploymentYAML
+
+	return common.DeployManagedResourceForSeed(ctx, c, managedResourceControlName, namespace, false, resources)
+}
+
+var (
+	crdTemplate                         *template.Template
+	versionConstraintK8sGreaterEqual112 *semver.Constraints
+)
+
+func init() {
+	var err error
+
+	crdTemplate, err = template.New("crd").Parse(crdTmpl)
+	utilruntime.Must(err)
+
+	versionConstraintK8sGreaterEqual112, err = semver.NewConstraint(">= 1.12")
+	utilruntime.Must(err)
+}
+
+const crdTmpl = `apiVersion: apiextensions.k8s.io/v1beta1
```

Comment:
Can we use the concrete type here instead of a template?

Additionally v1beta1 is deprecated -> we need to keep that in mind as they should be removed in one of the coming versions.
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.16.md#deprecations-and-removals

Reply:
No, I won't use Go types for this hundred lines long CRD definition, sorry.

>  we need to keep that in mind as they should be removed in one of the coming versions.

Yes, we will anyways have to go over the whole code when we drop support for a Kubernetes version.